frontend/src/components/playground/TransformationItem.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import { useState, useRef, useCallback } from "react";
+import { SaveState, SaveStatus } from "@/types/SaveStatus";
+import BoardItem from "@cloudscape-design/board-components/board-item";
+import Header from "@cloudscape-design/components/header";
+import Button, { ButtonProps } from "@cloudscape-design/components/button";
+import SpaceBetween from "@cloudscape-design/components/space-between";
+import Input from "@cloudscape-design/components/input";
+import { BoardProps } from "@cloudscape-design/board-components/board";
+import { Transformation } from "@/context/PlaygroundContext";
+import { boardItemI18nStrings } from "./boardItemI18nStrings";
+import AceEditorComponent from "./AceEditorComponent";
+import { usePlaygroundActions } from "@/hooks/usePlaygroundActions";
+import SaveStatusIndicator from "@/components/playground/SaveStatusIndicator";
+
+interface TransformationItemProps {
+  item: BoardProps.Item<Transformation>;
+  onRemove: (id: string) => void;
+}
+
+export default function TransformationItem({
+  item,
+  onRemove,
+}: Readonly<TransformationItemProps>) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editName, setEditName] = useState(item.data.name);
+  const [saveStatus, setSaveStatus] = useState<SaveState>({
+    status: SaveStatus.UNSAVED,
+    savedAt: null,
+    errors: [],
+  });
+  const { updateTransformation } = usePlaygroundActions();
+  const formatCodeRef = useRef<(() => void) | null>(null);
+
+  const handleSaveStatusChange = useCallback((status: SaveState) => {
+    setSaveStatus(status);
+  }, []);
+
+  const handleEditNameChange = (
+    event: CustomEvent<ButtonProps.ClickDetail>,
+  ) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (isEditing) {
+      if (editName.trim()) {
```

Comment:
Nit: what do you think about 'pre-trim'ing the name, leading/trailing whitespace is confusing on identifiers

Reply:
This particular check is basically just "is it empty", so I'm not going to change this, but I agree with you that preserving leading/trailing whitespace is a bad idea, so I've trimmed it in the `update` call on the line below.